qiskit_nature/second_q/mappers/tapered_qubit_mapper.py

```diff
@@ -0,0 +1,197 @@
+                operators.
+        """
+        super().__init__()
+        self.mapper: QubitMapper = mapper
+        self.z2symmetries = z2symmetries
+
+    @classmethod
+    def from_problem(
+        cls, mapper: QubitMapper, problem: "ElectronicStructureProblem"
+    ) -> "TaperedQubitMapper":
+        # implies the previous z2symmetries == "auto" case
+        # extract problem.hamiltonian and problem.symmetry_sector_locator
+        """Builds a ``TaperedQubitMapper`` from one of the other mappers and from a specific problem.
+        This simplifies the identification of the Pauli operator symmetries and of the symmetry sector
+        in which lies the solution of the problem.
+
+        Args:
+            mapper: ``QubitMapper`` object implementing the mapping of second quantized operators to
+                Pauli operators.
+            problem: A class encoding a problem to be solved.
+
+        Return:
+            A ``TaperedQubitMapper`` with pre-built symmetry specifications.
+        """
+
+        qubit_op, _ = problem.second_q_ops()
+        mapped_op = mapper.map(qubit_op).primitive
+        z2_symmetries = Z2Symmetries.find_z2_symmetries(mapped_op)
+        tapering_values = problem.symmetry_sector_locator(z2_symmetries, mapper)
+        z2_symmetries.tapering_values = tapering_values
+        return TaperedQubitMapper(mapper, z2_symmetries)
+
+    def _map_clifford_single(self, second_q_op: SparseLabelOp) -> SparsePauliOp:
+        mapped_op = self.mapper.map(second_q_op).primitive
+        converted_op = self.z2symmetries.convert_clifford(mapped_op)
+        return converted_op
+
+    def _symmetry_reduce_clifford_single(
+        self, converted_op: SparsePauliOp, check_commutes: bool = True
+    ) -> None | SparsePauliOp:
+
+        if self.z2symmetries.is_empty() or self.z2symmetries.tapering_values is None:
+            return converted_op
+        elif check_commutes:
+            logger.debug("Checking operator commute with symmetries:")
+            converted_symmetries = self.z2symmetries._sq_paulis
+            commutes = TaperedQubitMapper._check_commutes(converted_symmetries, converted_op)
+
+            if not commutes:
+                return None
+
+        tapered_op = self.z2symmetries.taper_clifford(converted_op)
+        cast(SparsePauliOp, tapered_op)
+        return tapered_op
+
+    def _map_single(self, second_q_op: SparseLabelOp) -> PauliSumOp | None:
+        converted_op = self._map_clifford_single(second_q_op)
+        tapered_op = self._symmetry_reduce_clifford_single(converted_op)
+
+        returned_op = PauliSumOp(tapered_op) if isinstance(tapered_op, SparsePauliOp) else None
+        return returned_op
+
+    def map_clifford(
+        self,
+        second_q_ops: SparseLabelOp | ListOrDictType[SparseLabelOp],
+    ) -> PauliSumOp | ListOrDictType[PauliSumOp]:
+        """Maps a second quantized operator or a list, dict of second quantized operators based on
+        the current mapper.
```

Comment:
This could be explained better. I think you should highlight the 2 steps taken here:
1. using the internal mapper, the operators are mapped
2. then, using the internal Z2Symmetries, `convert_clifford` is applied